include/glow/Support/Error.h

```diff
@@ -16,40 +16,219 @@
 #ifndef GLOW_SUPPORT_ERROR_H
 #define GLOW_SUPPORT_ERROR_H
 
+#include <cassert>
+#include <memory>
 #include <mutex>
 #include <type_traits>
 
-#include "llvm/Support/Error.h"
-#include "llvm/Support/FormatVariadic.h"
-
 #include <glog/logging.h>
 
+/// NOTE: please only use code and macros that resides outside of the detail
```

Comment:
Are most of the changes in this file "inspired" by LLVM's Error.h, i.e. is it pretty-much a copy of the LLVM's code with minor changes? Is there any possibility to change LLVM's behavior without literally re-defining all these types? Probably not. But it is annoying to have so much boilerplate code ;-(

Reply:
@opti-mix yeah it is overall pretty similar to LLVM's Error and Expected with a changes like abstracting out the state checking. The thing that it does is pretty dead simple so there isn't a lot of room for creative implementations.

As I mentioned in person, I would actually like to design this a bit differently, have only Expected<T> and model the Error use case with a template specialization for Expected<void> so we only have a single type everywhere. However for the sake of sanity, I think we should switch to a near drop-in replacement first and then consider later whether we want to go in a different direction like this instead of attempting to do everything at once.

Because llvm::Error's checking is disabled at compile time I don't think there is a clean and easy way to turn this back on by wrapping it externally and removing our dependency on llvm::Errors allows us the freedom to change things later as mentioned above.